osm2gtfs/creators/trips_creator.py

```diff
@@ -13,32 +19,301 @@ def __repr__(self):
         return rep
 
     def add_trips_to_feed(self, feed, data):
-        raise NotImplementedError("Should have implemented this")
+        """
+        This function generates and adds trips to the GTFS feed.
+
+        It is the place where geographic information and schedule is
+        getting joined to produce a routable GTFS.
+        """
+        all_trips_count = 0
+
+        # Go though all lines
+        for line_id, line in data.routes.iteritems():
+
+            print("\nGenerating schedule for line: [" + str(
+                line_id) + "] - " + line.name)
+
+            # Loop through it's itineraries
+            itineraries = line.get_itineraries()
+            for itinerary in itineraries:
+                trips_count = 0
+
+                # Verify data before proceeding
+                if self._verify_data(data.schedule, line, itinerary):
+
+                    # Prepare itinerary's trips and schedule
+                    prepared_trips = self._prepare_trips(feed, data.schedule,
+                                                         itinerary)
+
+                    # Add itinerary shape to feed.
+                    shape_id = self._add_shape_to_feed(
+                        feed, "relation/" + str(itinerary.osm_id), itinerary)
+
+                    # Add trips of each itinerary to the GTFS feed
+                    for trip_builder in prepared_trips:
+
+                        trip_builder['all_stops'] = data.get_stops()
+                        trips_count += self._add_itinerary_trips(
+                            feed, itinerary, line, trip_builder, shape_id)
+
+                # Print out status messge about added trips
+                print(" Itinerary: [" + str(itinerary.route_id) + "] " +
+                      itinerary.to.encode("utf-8") + " (added " + str(
+                          trips_count) + " trips, serving " + str(
+                              len(itinerary.get_stops())) + " stops) - " +
+                      itinerary.osm_url)
+                all_trips_count += trips_count
+
+        print("\nTotal of added trips to this GTFS: " +
+              str(all_trips_count) + "\n\n")
+        return
+
+    def _prepare_trips(self, feed, schedule, itinerary):
+        """
+        Prepare information necessary to generate trips
 
-    @staticmethod
-    def interpolate_stop_times(trip):
+        :return trips: List of different objects
         """
-        interpolate stop_times, because Navitia does not handle this itself by now
+
+        # Define a list with service days of given itinerary.
+        services = []
+        for trip in schedule['lines'][itinerary.route_id]:
+            input_fr = trip["from"]
+            input_to = trip["to"]
+            if input_fr == itinerary.fr and input_to == itinerary.to:
```

Comment:
we should support also `via` at this point and add the respective field to `itinerary` (see my change in the documentation of [Schedule](https://github.com/grote/osm2gtfs/wiki/Schedule#via))

Reply:
Done. See #101